crates/wasi-nn/src/backend/winml.rs

```diff
@@ -226,3 +205,180 @@ fn convert_i64(i: i64) -> Result<u32, BackendError> {
         anyhow::anyhow!("unable to convert dimension to u32: {d}").into()
     })
 }
+
+// Convert from wasi-nn tensor to WinML tensor.
+fn to_inspectable(tensor: &Tensor, shape: IVectorView<i64>) -> Result<IInspectable, Error> {
+    match tensor.ty {
+        crate::wit::types::TensorType::Fp16 => unsafe {
+            let data = std::slice::from_raw_parts(
+                tensor.data.as_ptr() as *const f32,
+                tensor.data.len() / size_of::<f32>(),
+            );
+            TensorFloat16Bit::CreateFromArray(&shape, data)?.cast::<IInspectable>()
+        },
+        crate::wit::types::TensorType::Fp32 => unsafe {
+            let data = std::slice::from_raw_parts(
+                tensor.data.as_ptr() as *const f32,
+                tensor.data.len() / size_of::<f32>(),
+            );
+            TensorFloat::CreateFromArray(&shape, data)?.cast::<IInspectable>()
+        },
+        crate::wit::types::TensorType::I64 => unsafe {
+            let data = std::slice::from_raw_parts(
+                tensor.data.as_ptr() as *const i64,
```

Comment:
```suggestion
                tensor.data.as_ptr().cast<i64>,
```

These days clippy tells me this is the preferred way to do this.

Reply:
Replaced with `cast`.